### PR TITLE
docs: Kotlin Conversion

### PR DESCRIPTION
**Pull request by david-allison-1**
Better to write this than to explain it. Reduces the 🚌 factor.

https://github.com/ankidroid/Anki-Android/pull/9480

Preview this PR: https://github.com/david-allison-1/Anki-Android/blob/c46ac3dd99696527b9d80cdee45e66eb3449f98d/docs/kotlin-migration.md

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
Nice!

File: docs/kotlin-migration.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+#### Tests/Android Tests
+
+Set `def source = Source.TEST/ANDROID_TEST` and keep `className` in the form: `/com/ichi2/...`
+
+### Workaround (second commit)
+
+Reset the path in `kotlinMigration.gradle` to the empty string. The path is no longer required due to the file conversion commit.
+
+## References
```
Comment (codingtosh) [inline docs/kotlin-migration.md:55]: Maybe add an example PR here, 
https://github.com/ankidroid/Anki-Android/pull/9738/commits

File: docs/kotlin-migration.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## Conversion
+
+We have developed a nonstandard process to convert existing files to Kotlin. Rationale is below
+
+## Process
+
+* Select the `.java` file in Android Studio and use the `Convert Java File to Kotlin File` functionality
+* Commit, selecting `Extra commit for java > kt renames`
+* Git rebase the first commit (`rename .java to .kt`)
```
Comment (codingtosh) [inline docs/kotlin-migration.md:13]: I would elaborate upon this a little. Something similar to:

Open the Git tab from the bottom left(Alt + 9) > Right click on the bottom commit (rename .java to .kt) > Click on Interactively Rebase from here > Right click on the top commit > Stop To Edit (Alt + E) -> Start Rebasing

File: docs/kotlin-migration.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+* Add the source-root relative path of the file to `kotlinMigration.gradle`
+* `File - Sync Project with Gradle Files`, then `Build - Rebuild` to ensure the commit builds
```
Comment (codingtosh) [inline docs/kotlin-migration.md:15]: Nit, ">" instead of "-" here and everywhere else
Example: Build > Rebuild

**Review by codingtosh — CHANGES_REQUESTED** [review]
LGTM @david-allison-1 ,
just a few nits.

**Comment by Arthur-Milchior** [comment]
I love it. Thanks forgot finding a way to deal correctly with git history

**Pull request merged**